lib/src/signed_video_h26x_sign.c

```diff
@@ -237,7 +237,7 @@ generate_sei_nalu(signed_video_t *self, uint8_t **payload, uint8_t **payload_sig
     if ((self->max_sei_payload_size > 0) && (payload_size > self->max_sei_payload_size) &&
         (mandatory_tags_size > 0)) {
       // Fallback to GOP-level signing
-      payload_size -= document_size;
+      payload_size -= mandatory_tags_size;
       self->gop_info->list_idx = -1;  // Reset hash list size to exclude it from TLV
       mandatory_tags_size =
           tlv_list_encode_or_get_size(self, mandatory_tags, num_mandatory_tags, NULL);
```

Comment:
On the line below, you are now adding the the `optional_tags_size`. This will make the payload_size incorrect.